src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -615,7 +639,7 @@ private AbstractFrame getClassFrameWhereViolationIsFound(DetailAST ast) {
         final DetailAST prevSibling = ast.getPreviousSibling();
         if (variableDeclarationFrameType == FrameType.CLASS_FRAME
                 && !validateOnlyOverlapping
-                && prevSibling == null
+                && (prevSibling == null || EXPR_TOKENS.contains(ast.getParent().getType()))
```

Comment:
There is similar code 16 lines down. Does that need to be changed too?

Reply:
I tried to remove it and it broke this test case:

```
   private String field1;
    void foo8(Long field1) {
        field1 += field1; // violation
    }
```
with config validateOnlyOverlapping = false and other options = default

I am puzzled how it should be checked because it have 3 options (add this to both or add to only one) and it is not clear what is correct.

Can you clarify it?

Reply:
> I tried to remove it 

Are you talking about removing the code 16 lines down? Why remove it when this item was asking about updating it to be insync with the changes we are making here?

Reply:
@rnveach nevemind. The thing is that this condition (prevSibling == null) in condition below is responsible for test case
```
private String field1;
    void foo8(Long field1) {
        field1 += field1; // violation
    }
```
Can you clarify for this case, what should be a violation? It is not clear for me.

Reply:
> The thing is that this condition (prevSibling == null) in condition below is responsible for test case

Just so we are clear, I am not asking to remove that condition for this PR.

> Can you clarify for this case, what should be a violation? It is not clear for me.

@romani ping. Can you clarify this line.
I want to say that if the same name is used in both places and the name represents a field and a non-field, we request the the name on the left of the assignment have a `this.` as it most likely should represent the field and the non-field is blocking the correct assignment.

Reply:
@rnveach I created a diff report for this condition.
Base branch is this one, patch branch contains same check for condition below
https://strkkk.github.io/7066_3/index.html

It looks like adding this condition will cause a lot of false-positives, e.g.
https://strkkk.github.io/7066_3/lombok-ast/xref/checkstyle/contribution/checkstyle-tester/repositories/lombok-ast/src/ecjTransformer/lombok/ast/ecj/EcjTreeConverter.java.html#L230

Reply:
I agree, it looks like changing that code is a bad idea.
So we won't do this. Thanks for confirming.

Reply:
I running late .... sorry.
Please read documentation of this Check on such cases. I agree it is gray area of this Check.
Please try to analyze what is now and what it should be.

I will be back after I unblock eclipse-cs project, sorry.